src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -1 +1,112 @@
-pub struct BusinessLogicSyscallHandler;
+#![allow(dead_code)]
+
+use cairo_rs::{types::relocatable::Relocatable, vm::vm_core::VirtualMachine};
+use felt::Felt252;
+
+use crate::{
+    business_logic::{
+        fact_state::state::ExecutionResourcesManager,
+        state::{
+            contract_storage_state::ContractStorageState,
+            state_api::{State, StateReader},
+        },
+    },
+    core::errors::syscall_handler_errors::SyscallHandlerError,
+    utils::Address,
+};
+
+use super::{
+    syscall_handler::SyscallHandler,
+    syscall_info::get_syscall_size_from_name,
+    syscall_request::{FromPtr, SyscallRequest},
+    syscall_response::SyscallResponse,
+};
+
+pub struct BusinessLogicSyscallHandler<'a, T: Default + State + StateReader> {
+    pub(crate) resources_manager: ExecutionResourcesManager,
+    pub(crate) expected_syscall_ptr: Relocatable,
+    pub(crate) starknet_storage_state: ContractStorageState<'a, T>,
+}
+
+impl<'a, T: Default + State + StateReader> BusinessLogicSyscallHandler<'a, T> {
+    fn new(
+        state: &'a mut T,
+        contract_address: Address,
+        resources_manager: ExecutionResourcesManager,
+        expected_syscall_ptr: Relocatable,
+    ) -> Self {
+        let starknet_storage_state = ContractStorageState::new(state, contract_address);
+        Self {
+            resources_manager,
+            expected_syscall_ptr,
+            starknet_storage_state,
+        }
+    }
+
+    fn increment_syscall_count(&mut self, syscall_name: &str) {
+        self.resources_manager
+            .increment_syscall_counter(syscall_name, 1);
+    }
+
+    fn _storage_write(&mut self, key: Felt252, value: Felt252) {
```

Comment:
please rename the function to `syscall_storage_write` to avoid using underscore. I just this in order to keep consistency with the changes to this things that we already did in main.